benchmarks/float8/bench_linear_float8.py

```diff
@@ -112,137 +81,93 @@ def main(
     input_bias = False
-    if fast_accum_filter is not None:
-        use_fast_accum = [fast_accum_filter]
-    else:
-        use_fast_accum = [True, False]
     if shape_name_filter is not None:
         k = shape_name_filter
         name_to_shapes = ((k, v) for (k, v) in name_to_shapes if k == shape_name_filter)
     experiment_list: List[Experiment] = []
     dtype = torch.bfloat16
-    for idx, (fast_accum, (name, (M, K, N))) in enumerate(
-        tqdm(list(product(use_fast_accum, name_to_shapes)))
-    ):
+    for idx, (name, (M, K, N)) in enumerate(tqdm(name_to_shapes)):
         if n_limit is not None and idx >= n_limit:
             break
-        linear_ref = torch.nn.Linear(K, N, bias=input_bias).to(
-            device=device, dtype=dtype
-        )
+        m_ref, input_tensors = subgraph_name_to_subgraph_and_inputs(
+            model_type, mkn_override=(M, K, N))
 
-        linear_float8 = Float8Linear.from_float(
-            copy.deepcopy(linear_ref),
-            config=config,
-        )
-        scaling_repr = f"{linear_float8.scaling_type_repr()},{linear_float8.scaling_granularity_repr()}"
+        # linear_ref = nn.Sequential(nn.Linear(K, N, bias=input_bias)).to(
+        #     device=device, dtype=dtype
+        # )
+        # input_tensor = torch.randn(M, K, device=device, dtype=dtype, requires_grad=True)
 
-        if fast_accum:
-            linear_float8.forward_config = ScaledMMConfig(False, True, False)
-        else:
-            linear_float8.forward_config = ScaledMMConfig(False, False, False)
+        m_float8 = copy.deepcopy(m_ref)
+        convert_to_float8_training(m_float8, config=config)
 
-        input_tensor = torch.randn(M, K, device=device, dtype=dtype, requires_grad=True)
-        ref_forw_backward = lambda: linear_ref(input_tensor).sum().backward()
+        def ref_forw_backward(*input_tensors):
+            return m_ref(*input_tensors).sum().backward()
 
-        def float8_forw_backward():
+        def float8_forw_backward(*input_tensors):
             if linear_requires_sync(config):
-                sync_float8_amax_and_scale_history(linear_float8)
-            linear_float8(input_tensor).sum().backward()
-
-        def n_times(n, fn, *args, **kwargs):
-            def wrapper(*args, **kwargs):
-                for _ in range(n):
-                    fn(*args, **kwargs)
-
-            return wrapper
-
-        REPEAT_N = 100
-
-        ref_forw_backward = n_times(REPEAT_N, ref_forw_backward)
-        float8_forw_backward = n_times(REPEAT_N, float8_forw_backward)
+                sync_float8_amax_and_scale_history(m_float8)
+            return m_float8(*input_tensors).sum().backward()
 
         if compile:
             ref_forw_backward = torch.compile(ref_forw_backward)
             float8_forw_backward = torch.compile(float8_forw_backward)
 
-        for _ in range(5):
-            ref_forw_backward()
-            float8_forw_backward()
+        n_warmup = 2
```

Comment:
nit: maybe move else if you think people will change warmup steps, or just write
# Warmups